### PR TITLE
Chore: Rewrite git ignores for yarn related files

### PR DESCRIPTION
**Pull request by jackw**
**What is this feature?**

It seems our dependency bots no longer commit yarn updates to .yarn/releases and I've had bother trying to commit to .yarn/patches . I think it's related to [this change](https://github.com/grafana/grafana/commit/157ea31b03fe9b9ea6bf7d43ae15b4e9f7a57dff#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947).

**Why do we need this feature?**

So dependabot can update yarn without us needing to intervene.

**Who is this feature for?**

Grafana maintainers

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by mckn — APPROVED** [review]
LGTM!

File: .gitignore [hunk 1]
```diff
@@ -17,13 +17,11 @@ vendor/
...
-!.yarn/patches
-!.yarn/releases
-!.yarn/plugins
-!.yarn/sdks
-!.yarn/versions
+# yarn
+.yarn/cache
+.yarn/unplugged
+.yarn/build-state.yml
+.yarn/install-state.gz
```
Comment (torkelo) [inline .gitignore:24]: @jackw 

with this change I now have 

```
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .yarn/.cache/
        .yarn/sdks/
```

Comment (torkelo) [reply .gitignore:24]: but seem like I can delete those, thought sdks would come back after `yarn dlx @yarnpkg/sdks vscode`

Comment (jackw) [reply .gitignore:24]: @torkelo pretty sure we don't [use sdks](https://yarnpkg.com/getting-started/editor-sdks) any more with yarn berry as we stopped using PnP a while back. Do you have them referencing your vscode settings somewhere? 🤔

Comment (jackw) [reply .gitignore:24]: [This PR that switched us off PnP](https://github.com/grafana/grafana/pull/79947) removed the sdks. The .cache directory should be fine to delete as I think that's where PnP stores the .zip files.

Comment (jackw) [reply .gitignore:24]: If we think it's gonna cause widespread bother we can add them as ignored files.

**Pull request merged**